spanner: many tests failed

Many tests failed at the same time in this package.

* I will close this issue when there are no more failures in this package _and_
  there is at least one pass.
* No new issues will be filed for this package until this issue is closed.
* If there are already issues for individual test cases, I will close them when
  the corresponding test passes. You can close them earlier, if you prefer, and
  I won't reopen them while this issue is still open.

Here are the tests that failed:
* TestPartitionRoundTrip
* TestBROTIDRoundTrip
* TestPartitionQuery_QueryOptions/Query_level_has_highest_precedence
* TestPartitionQuery_Parallel
* TestClient_Single
* TestClient_Single_Unavailable
* TestClient_Single_InvalidArgument
* TestClient_Single_SessionNotFound
* TestClient_Single_Read_SessionNotFound
* TestClient_Single_ReadRow_SessionNotFound
* TestClient_Single_RetryableErrorOnPartialResultSet
* TestClient_Single_NonRetryableErrorOnPartialResultSet
* TestClient_Single_NonRetryableInternalErrors
* TestClient_Single_DeadlineExceeded_NoErrors
* TestClient_Single_DeadlineExceeded_WithErrors
* TestClient_Single_ContextCanceled_noDeclaredServerErrors
* TestClient_Single_ContextCanceled_withDeclaredServerErrors
* TestClient_Single_QueryOptions/Client_level


-----
commit: d5e59f14debc5b68e0ba7054f2a086c43dc4631a
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/0ca32bc0-7313-4253-a45b-7d7878b214a9), [Sponge](http://sponge2/0ca32bc0-7313-4253-a45b-7d7878b214a9)
status: failed

Looking at the most recent post from flaky-bot it looks like there is really only one test failure left:

```bash
--- FAIL: TestSessionCreationIsDistributedOverChannels (0.03s)
panic: reflect: call of reflect.Value.Elem on zero Value [recovered]
	panic: reflect: call of reflect.Value.Elem on zero Value

goroutine 6803 [running]:
testing.tRunner.func1(0xc002cfb200)
	/usr/local/go/src/testing/testing.go:792 +0x6a7
panic(0x10573c0, 0xc000440560)
	/usr/local/go/src/runtime/panic.go:513 +0x1b9
reflect.Value.Elem(0x0, 0x0, 0x0, 0x116e41a, 0x8, 0x0)
	/usr/local/go/src/reflect/value.go:805 +0x39c
cloud.google.com/go/spanner.getSessionsPerChannel(0xc0039bad00, 0x0)
	/tmpfs/src/google-cloud-go/spanner/session_test.go:1981 +0x1f5
cloud.google.com/go/spanner.TestSessionCreationIsDistributedOverChannels(0xc002cfb200)
	/tmpfs/src/google-cloud-go/spanner/session_test.go:1935 +0x24f
testing.tRunner(0xc002cfb200, 0x11a4c18)
	/usr/local/go/src/testing/testing.go:827 +0x163
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:878 +0x65a
FAIL	cloud.google.com/go/spanner	114.275s
```

I think this is unrelated from the backend changes in the gapic layer, please let me know if I am wrong though!

--- a/spanner/session_test.go
+++ b/spanner/session_test.go
@@ -1978,7 +1978,8 @@ func getSessionsPerChannel(sp *sessionPool) map[string]int {
 		// Get the pointer to the actual underlying gRPC ClientConn and use
 		// that as the key in the map.
 		val := reflect.ValueOf(s.client).Elem()
-		connPool := val.FieldByName("connPool").Elem().Elem()
+		internalClient := val.FieldByName("internalClient").Elem().Elem()
+		connPool := internalClient.FieldByName("connPool").Elem().Elem()
 		conn := connPool.Field(0).Pointer()
 		key := fmt.Sprintf("%v", conn)
 		sessionsPerChannel[key] = sessionsPerChannel[key] + 1

